chromadb/api/models/Collection.py

```diff
@@ -71,17 +72,23 @@ def add(
             ValueError: If you provide an id that already exists
 
         """
-        (
+        embedding_set = self._process_add_request(
             ids,
             embeddings,
             metadatas,
             documents,
+            images,
             uris,
-        ) = self._validate_and_prepare_embedding_set(
-            ids, embeddings, metadatas, documents, images, uris
         )
 
-        self._client._add(ids, self.id, embeddings, metadatas, documents, uris)
+        self._client._add(
+            embedding_set["ids"],
+            self.id,
+            cast(Embeddings, embedding_set["embeddings"]),
```

Comment:
I think we only have this cast because we re-use the `EmbeddingSet` type for a case where we know for sure `embeddings` can't be `None`. Not sure it's worth the additional type, but it's worth discussing.